relay-server/src/actors/outcome.rs

```diff
@@ -279,10 +279,12 @@ pub struct TrackRawOutcome {
     /// The client ip address.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     remote_addr: Option<String>,
+    /// The source of the outcome (which Relay sent it)
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    source: Option<String>,
 }
-
-impl From<&TrackOutcome> for TrackRawOutcome {
-    fn from(msg: &TrackOutcome) -> Self {
+impl TrackRawOutcome {
+    fn from_outcome(msg: &TrackOutcome, config: &Config) -> Self {
```

Comment:
I know this was carried over from the previous implementation -- yet, would like to revisit: Can we make this consume `msg: TrackOutcome` rather than pass a reference and clone internals? This would avoid the string clone for `source`.

Reply:
Changed to passing by value since it makes more sense.
From a practical point of view it doesn't change anything since all values used from `TrackOutcome` are either value types or processed in some way.

The source does not come from the outcome but from the configuration ( all `TrackOutcome` messages are generated on the current Relay and do not carry a source, so the source is taken directly from the relay configuration).

Reply:
You're right, I misread the source when scanning through the PR. Sorry about that.